src/S3Writer.php

```diff
@@ -85,15 +85,7 @@ public function execute(string $sourcePath): void
                     /** @var ClientException $previous */
                     $previous = $e->getPrevious();
                     if ($previous->getResponse()) {
-                        throw new UserException(
-                            $previous->getResponse()->getStatusCode()
-                            . " "
-                            . $previous->getResponse()->getReasonPhrase()
-                            . " ("
-                            . $e->getAwsErrorCode()
-                            . ")\n"
-                            . $previous->getResponse()->getBody()->__toString()
-                        );
+                        throw S3WriterException::factory($e);
```

Comment:
osobně spíš používám static constructor. Tj. např. S3WriterException::fromS3Exception(). Akorát jsem myslel, že zabalíš celou tu logiku. Jakože "z s3 exceptiony (libovolné) vyrobím S3WriterException takto"... (a do toho zabalit celou tu logiku včetně kontrol.

Reply:
To je ale totéž, liší se jen název. Nebo mi něco ještě uniká? Celou logiku tam teda zabalím.

Reply:
Neuniká ti tam nic. Jen pak máš možnost mít třeba ještě `::fromConnectException()` na jiném místě.

Reply:
jasný! to je super